### PR TITLE
Overriding/OverriddenMethodFilter

### LINKED ISSUE #274: add support for override analysis
We have a transformation that requires a method that gives the overridden method.

```
CtMethod {
  // returns the overridden method, null if none 
  CtMethod getOverridenMethod()

}
```

### PR DESCRIPTION
**Pull request by GerardPaligot**
Closes #274

### CODE REVIEW

File: src/main/java/spoon/reflect/visitor/filter/OverridingMethodFilter.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import spoon.reflect.reference.CtExecutableReference;
+import spoon.reflect.visitor.Filter;
+
+/**
+ * Gets all overriding method from the method given.
+ */
+public class OverridingMethodFilter implements Filter<CtMethod<?>> {
+	private final CtExecutableReference<?> executableReference;
+
+	public OverridingMethodFilter(CtExecutableReference<?> executableReference) {
```
Comment (monperrus) [inline src/main/java/spoon/reflect/visitor/filter/OverridingMethodFilter.java]: Take as parameter a CtMethod (and not a reference)?

avoid the getReference in usage (cf test) with encapsulatio

File: src/main/java/spoon/reflect/visitor/filter/OverriddenMethodFilter.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import spoon.reflect.reference.CtExecutableReference;
+import spoon.reflect.visitor.Filter;
+
+/**
+ * Gets all overridden method from the method given.
+ */
+public class OverriddenMethodFilter implements Filter<CtMethod<?>> {
+	private final CtExecutableReference<?> executableReference;
+
+	public OverriddenMethodFilter(CtExecutableReference<?> executableReference) {
```
Comment (monperrus) [inline src/main/java/spoon/reflect/visitor/filter/OverriddenMethodFilter.java]: idem

File: src/main/java/spoon/reflect/visitor/filter/OverridingMethodFilter.java (hunk 1 above)
Comment (GerardPaligot) [inline src/main/java/spoon/reflect/visitor/filter/OverridingMethodFilter.java]: I adopted the same logic of `InvocationFilter` for this: https://github.com/INRIA/spoon/blob/master/src/main/java/spoon/reflect/visitor/filter/InvocationFilter.java

Comment (monperrus) [inline src/main/java/spoon/reflect/visitor/filter/OverridingMethodFilter.java]: good idea to have a regular design. but I would rather update
CtInvocationFilter with a new constructor with a CtMethod.

Comment (GerardPaligot) [inline src/main/java/spoon/reflect/visitor/filter/OverridingMethodFilter.java]: We don't need a `CtMethod` in `InvocationFilter`. Why we should have a constructor useless?

**Comment by monperrus** [comment]
> We don't need a CtMethod in InvocationFilter

We need it for regaularity in the API. And this will be useful for users in the future.

**Comment by GerardPaligot** [comment]
@monperrus PR ok.

**Pull request merged**